### DISCUSSION TITLE
Is it possible to fill the event window with synthetic (copied) events?

### DISCUSSION THREAD

**Discussion by nicnike** [microsoft/perfview#1595] [Q&A]
Hi, 

is it possible to load the eventwindow additionally with copied or slightly modified events from the payload? (via Extension)

Some of our customized events have very long sql queries in their payload and we would like to shorten/modify them and then display them as a copy in addition.

Any support/input would be nice :)

**Comment by brianrob** [comment]
@nicnike, as far as I can tell, there isn't anything that will allow you to do this.  But something you can do is to create a PerfView extension (see `PerfView CreateExtensionProject`) in the command line help.  From there, you can access all events, and so you could open them in Excel, you could modify them and write them into the log, etc.  You also could choose not to use PerfView at all, but could write an app that uses TraceEvent to read your custom events, modify them they way you want, and then write them to a file, console, etc.
